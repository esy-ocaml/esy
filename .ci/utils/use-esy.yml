# steps to install esy globally

steps:
- script: "yarn global add @esy-nightly/esy@0.6.8-bbdfb8"
  displayName: "Install esy (via yarn)"
  condition: ne(variables['AGENT.OS'], 'Windows_NT')
  
- script: |
    yarn config set prefix C:\
    yarn global add @esy-nightly/esy@0.6.8-bbdfb8
  displayName: "Install esy (via yarn)"
  condition: eq(variables['AGENT.OS'], 'Windows_NT')

- bash: |
    YARN_BIN_DIR=$(yarn global bin)
    echo $YARN_BIN_DIR
    echo "##vso[task.setvariable variable=YARN_BIN_DIR]$YARN_BIN_DIR"
  displayName: "Setting $YARN_BIN_DIR"

- bash: |
    echo "##vso[task.setvariable variable=PATH]$PATH:$YARN_BIN_DIR" 
  displayName: "Ensure yarn install global tools are available on $PATH"
  condition: ne(variables['AGENT.OS'], 'Windows_NT')

- powershell: Write-Host "##vso[task.setvariable variable=PATH;]${env:PATH};${env:YARN_BIN_DIR}"
  displayName: "Ensure yarn install global tools are available on $PATH"
  condition: eq(variables['AGENT.OS'], 'Windows_NT')
