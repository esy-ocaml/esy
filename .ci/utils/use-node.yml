# steps to use node on agent

steps:
  - task: NodeTool@0
    displayName: "Use Node 15.x"
    inputs:
      versionSpec: 15.x
