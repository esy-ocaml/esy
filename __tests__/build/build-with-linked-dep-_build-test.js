/**
 * @flow
 */

import {defineTestCaseWithShell} from './utils';

defineTestCaseWithShell(
  'with-linked-dep-_build',
  `
    run esy build

    # package "dep" should be visible in all envs
    assertStdout "esy dep" "dep"
    assertStdout "esy b dep" "dep"
    assertStdout "esy x dep" "dep"

    assertStdout "esy x with-linked-dep-_build" "with-linked-dep-_build"
  `,
);
