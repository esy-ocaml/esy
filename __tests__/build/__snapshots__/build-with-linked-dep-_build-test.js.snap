// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build with-linked-dep-_build 1`] = `
"RUNNING: esy config ls
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
RUNNING: esy build
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:bin needRebuildTarget <sandbox>/node_modules/.cache/_esy/build/bin/build
esy:bin:esyBuild execute
esy:build:dep@1.0.0 start building
esy:build:dep@1.0.0 build success
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:build-dependencies checking if dependencies are built
RUNNING: esy dep
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:bin needRebuildTarget <sandbox>/node_modules/.cache/_esy/build/bin/build
RUNNING: esy b dep
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:bin needRebuildTarget <sandbox>/node_modules/.cache/_esy/build/bin/build
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:build-dependencies checking if dependencies are built
RUNNING: esy x dep
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:bin needRebuildTarget <sandbox>/node_modules/.cache/_esy/build/bin/build
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:build-dependencies checking if dependencies are built
RUNNING: esy x with-linked-dep-_build
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:bin needRebuildTarget <sandbox>/node_modules/.cache/_esy/build/bin/build
esy:bin no .esyrc found
esy:bin prefix: <testRoot>/esy
esy:build-dependencies checking if dependencies are built"
`;
